File: 637 - Average of Levels in Binary Tree.java
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Deque;

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Double> averageOfLevels(TreeNode root) {
        Deque q = new ArrayDeque<>();
        if (root != null) {
            q.offer(root);
        }
        List<Double> res = new ArrayList<>();
        int len;
        double sum;
        while (q.size() != 0) {
            len = q.size();
            sum = 0;
            for (int i = 0; i < len; i++) {
                root = (TreeNode) q.poll();
                sum += root.val;
                if (root.left != null) q.offer(root.left);
                if (root.right != null) q.offer(root.right);
            }
            res.add(sum / len);
        }
        return res;
    }
}

File: 637 - Average of Levels in Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var averageOfLevels = function(root) {
    let q = [], res = [];
    if (root != null) {
        q.push(root);
    }
    let sum, node, len;
    while (q.length != 0) {
        len = q.length;
        sum = 0;
        for (let i = 0; i < len; i++) {
            node = q.shift();
            sum += node.val;
            if (node.left != null) q.push(node.left);
            if (node.right != null) q.push(node.right);
        }
        res.push(sum / len);
    }
    return res;
};